src/miner.cpp

```diff
@@ -346,6 +351,8 @@ CBlockTemplate* CreateNewBlock(const CScript& scriptPubKeyIn)
         txNew.vout.resize(1);
         txNew.vout[0].scriptPubKey = scriptPubKeyIn;
         txNew.vout[0].nValue = GetBlockSubsidy(nHeight, chainparams.GetConsensus());
+        // Set coinbase expiry to current block height
+        txNew.nBlockExpiry = nHeight;
```

Comment:
I had it set to nBlockExpiry + 1, then changed it back, thinking a coinbase should only be valid for a single block? Was unaware of this behavior of pools. Not sure what the best way to handle Coinbase txs is, did not include this detail in the ZIP for feedback (will go add it now).

Reply:
Ah, I see the confusion. As-defined in your ZIP, `nBlockExpiry` is the last height at which the transaction is valid. But in your `IsExpired()` function, `nBlockExpiry` is used as the first height at which the transaction expires, and thus `nBlockExpiry = nHeight` semantically would mean that the coinbase transaction is just-expired. We should fix one or the other.

Reply:
Good point that this could be a hassle for mining pools.

Since coinbase txs are already special, could specify that the expiry height is ignored for coinbase transactions.

We might want to specify that this field is set to no expiry... or say that it is a reserved field for now and should not used be for anything.

That gives us 32 bytes which could be used at a later day... maybe for a merkle root as part of a malleability fix.

Reply:
@str4d Ah I see, I will change the implementation to match the ZIP since that's what has been discussed.

@bitcartel  You mean set the expiry height on coinbase txs to a null value?